### ISSUE TITLE
Fails in iOS browser: unknown function run_lua

### ISSUE THREAD

**Issue by jamshid** [vvanders/wasm_lua#1]
Very nice, thanks. Any idea when iOS browsers will support this? Currently getting below error, even with iOS 10.3 Beta. 
```Assertion failed: "Cannot call unknown function run_lua (perhaps LLVM optimizations or closure removed it?)```
Does Apple purposely block the use of lua scripts?

**Comment by mohsen1** [comment]
Web Assembly is not supported in iOS Safari.

**Comment by derrohrbach** [comment]
Same problem in chrome on android :(

**Comment by vvanders** [comment]
Yeah, this is WASM only which works on a subset of browsers. Totally happy to take a PR for polyfill via asm.js if anyone is interested.

**Comment by jamshid** [comment]
Darn. Hopefully this FAQ answer at bottom of http://webassembly.org/docs/faq/ means Apple has agreed to (soon?) support Web Assembly in iOS web browsers, providing at least the same device api that javascript in Safari gets.

> Inside a browser you’ll get access to the same HTML5 and other browser-specific APIs which are also accessible through regular JavaScript. 

From reading around (e.g. https://news.ycombinator.com/item?id=12838884) it seems iOS apps can _use_ lua scripts bundled in the app, but an app is not allowed to download new scripts. 

Web browsers on iOS are only allowed to use Apple's javascript engine. Somehow that engine prevents lua / WASM from working. I guess it's debatable whether Apple does this for security reasons or to maintain control (i.e., no alternative app stores).

**Comment by xspager** [comment]
Take this with a grain of salt:

I changed the makefile to generate a asmjs fallback and it now works on Safari

```diff
diff --git a/makefile b/makefile
index e6fde80..e8d447b 100644
--- a/makefile
+++ b/makefile
@@ -1,7 +1,7 @@
 all: main.o

 main.o: main.c lua
-       emcc -Ilua-5.3.4/src main.c lua-5.3.4/src/liblua.a -s WASM=1 -O2 -o main.js -s EXPORTED_FUNCTIONS="['_run_lua']"
+       emcc -Ilua-5.3.4/src main.c lua-5.3.4/src/liblua.a -s BINARYEN=1 -s "BINARYEN_METHOD='native-wasm,asmjs'" -O2 -o main.js -s EXPORTED_FUNCTIONS="['_run_lua']"

 lua:
-       cd lua-5.3.4/src && make generic CC='emcc -s WASM=1'
\ No newline at end of file
+       cd lua-5.3.4/src && make generic AR='emar ru' RANLIB='emranlib' CC='emcc -s WASM=1'
```

**Comment by vvanders** [comment]
That's awesome, if you want to throw that up in a PR I'm happy to give it a test and merge it in if it looks good on this end.
